server/src/test/java/org/elasticsearch/search/geo/GeoBoundsGenericWriteableTests.java

```diff
@@ -0,0 +1,113 @@
+import org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.test.AbstractNamedWriteableTestCase;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class GeoBoundsGenericWriteableTests extends AbstractNamedWriteableTestCase<GenericNamedWriteable> {
+    NamedWriteableRegistry registry = new NamedWriteableRegistry(
+        List.of(new NamedWriteableRegistry.Entry(GenericNamedWriteable.class, GeoBoundingBox.class.getSimpleName(), GeoBoundingBox::new))
+    );
+
+    @Override
+    protected GeoBoundingBox createTestInstance() {
+        Rectangle box = GeoTestUtil.nextBox();
+        return new GeoBoundingBox(new GeoPoint(box.maxLat, box.minLon), new GeoPoint(box.minLat, box.maxLon));
+    }
+
+    @Override
+    protected GeoBoundingBox mutateInstance(GenericNamedWriteable instance) throws IOException {
+        assert instance instanceof GeoBoundingBox : "Expected GeoBoundingBox";
+        GeoBoundingBox geoBoundingBox = (GeoBoundingBox) instance;
+        double width = geoBoundingBox.right() - geoBoundingBox.left();
+        double height = geoBoundingBox.top() - geoBoundingBox.bottom();
+        double top = geoBoundingBox.top() - height / 4;
+        double left = geoBoundingBox.left() + width / 4;
+        double bottom = geoBoundingBox.bottom() + height / 4;
+        double right = geoBoundingBox.right() - width / 4;
+        return new GeoBoundingBox(new GeoPoint(top, left), new GeoPoint(bottom, right));
+    }
+
+    @Override
+    protected GenericNamedWriteable copyInstance(GenericNamedWriteable original, TransportVersion version) throws IOException {
+        return copyInstance(original, version, version);
+    }
+
+    /**
+     * Read and write with different serialization versions
+     */
+    protected GenericNamedWriteable copyInstance(
+        GenericNamedWriteable original,
+        TransportVersion writeVersion,
+        TransportVersion readVersion
+    ) throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            output.setTransportVersion(writeVersion);
+            output.writeGenericValue(original);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), getNamedWriteableRegistry())) {
+                in.setTransportVersion(readVersion);
+                return readGenericValue(in);
+            }
+        }
+    }
+
+    private GenericNamedWriteable readGenericValue(StreamInput in) throws IOException {
+        Object obj = in.readGenericValue();
+        if (obj instanceof GenericNamedWriteable result) {
+            return result;
+        }
+        throw new IllegalStateException("Read result of wrong type: " + obj.getClass().getSimpleName());
+    }
+
```

Comment:
This is no longer needed, we can just use the `copyInstance` implementation in the subclass.

```suggestion
```

(also needs a spotless run to clean up imports)

Reply:
acked, I think I can see a way to make this a little less special, but let's merge this and I'll follow up with a suggestion.